android/src/main/java/com/reactcommunity/rndatetimepicker/RNMaterialRangePicker.kt

```diff
@@ -0,0 +1,207 @@
+package com.reactcommunity.rndatetimepicker
+
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.core.util.Pair
+import androidx.fragment.app.FragmentManager
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.WritableNativeMap
+import com.google.android.material.datepicker.CalendarConstraints
+import com.google.android.material.datepicker.CalendarConstraints.DateValidator
+import com.google.android.material.datepicker.CompositeDateValidator
+import com.google.android.material.datepicker.DateValidatorPointBackward
+import com.google.android.material.datepicker.DateValidatorPointForward
+import com.google.android.material.datepicker.MaterialDatePicker
+import com.google.android.material.datepicker.MaterialPickerOnPositiveButtonClickListener
+import java.util.Calendar
+
+class RNMaterialRangePicker(
+  private val args: Bundle,
+  private val promise: Promise,
+  private val fragmentManager: FragmentManager,
+  private val reactContext: ReactApplicationContext
+) {
+  private var promiseResolved = false
+  private var rangePicker: MaterialDatePicker<Pair<Long, Long>>? = null
+  private var builder = MaterialDatePicker.Builder.dateRangePicker()
+
+  fun open() {
+    createRangePicker()
+    addListeners()
+    show()
+  }
+
+  private fun createRangePicker() {
+    setInitialDates()
+    setTitle()
+    setInputMode()
+    setButtons()
+    setConstraints()
+    setFullscreen()
+
+    rangePicker = builder.build()
+  }
+
+  private fun setInitialDates() {
+    var start: Long? = null
+    var end: Long? = null
+
+    if (args.containsKey(RNConstants.ARG_START_TIMESTAMP)) {
+      // override "value" so we can use the same constructor from RNDate
+      args.putLong(RNConstants.ARG_VALUE, args.getLong((RNConstants.ARG_START_TIMESTAMP)))
+      start = RNDate(args).timestamp()
+    }
+
+    if (args.containsKey(RNConstants.ARG_END_TIMESTAMP)) {
+      // override "value" so we can use the same constructor from RNDate
+      args.putLong(RNConstants.ARG_VALUE, args.getLong((RNConstants.ARG_END_TIMESTAMP)))
+      end = RNDate(args).timestamp()
+    }
+
+    val selection = Pair(start, end)
+    builder.setSelection(selection)
+  }
+
+  private fun setTitle() {
+    val title = args.getString(RNConstants.ARG_TITLE)
+    if (!title.isNullOrEmpty()) {
+      builder.setTitleText(args.getString(RNConstants.ARG_TITLE))
+    }
```

Comment:
```suggestion
    args.getString(RNConstants.ARG_TITLE)?.let {
      builder.setTitleText(it)
    }
```

more kotlin-y. You can use this pattern more across this file pls, e.g in setButtons.